pkg/pup/manifest.go

```diff
@@ -0,0 +1,127 @@
+package pup
+
+/* PupManifest represents a Nix installed process
+ * running inside the Dogebox Runtime Environment.
+ * These are defined in pup.json files.
+ */
+type PupManifestV1 struct {
```

Comment:
I thought the V1 was just explanatory for the gist, I assume the intent is for migration? If that's the case what about type-aliasing ? ie:

```go
type PupManifestV1 struct { ...

type PupManifest = PupManifestV1
```


etc..

That way we don't have to change the code everywhere but can update the aliases ?

Reply:
@tjstebbing Happy to rename it for now because yeah I copied it straight from the gist. I do think we'll eventually want to change the schema eventually once we break it in (so to speak) and we & the community build on it and figure out what is missing/needs changing, at which time we either:

- 1) Rename `PupManifest` back to `PupManifestV1`
- 2) Create a `PupManifestV2`, and alias `PupManifest` to that, though all references might still need to be updated (especially if the actual parsed interface is different, and not just the structure of the json file we're loading from).

Thoughts?

Reply:
Yeh makes sense :)